### PR TITLE
test: cover partial plant payloads

### LINKED ISSUE #212: Add regression coverage for partial plant payloads
## Context

The completed July 2026 live multi-region audit confirmed that partial plant payloads are normal API behavior, not an edge case.

Across **19 successful five-day responses** and **95 location-days**:

- 17 locations returned at least one plant entry with `code` present and `indexInfo` absent somewhere in the horizon;
- optional fields such as `inSeason`, `plantDescription`, `displayName`, and individual color channels were absent or partial in real responses;
- 9 locations had plant entries whose optional structure changed across forecast days;
- RC3 lost no plant keys and raised no coordinator-shaping errors;
- 0 invalid index values were observed;
- all 19 payloads were structurally compatible with RC3.

This issue locks the current tolerant behavior in with focused regression tests only.

## Goal

Protect the current shaping contract for real partial Google Pollen API plant entries without changing runtime behavior.

## Required tests

Add focused tests proving that a plant with a valid non-empty `code` remains usable when any combination of these optional fields is absent or partial.

### Missing `indexInfo`

- [ ] the plant remains present in `coordinator.data`;
- [ ] `source == "plant"`;
- [ ] code and stable sensor key are preserved;
- [ ] `value`, `category`, `description`, `color_hex`, and `color_rgb` are `None`;
- [ ] processing of other pollen types and plants continues.

### Missing plant metadata

- [ ] missing `inSeason` produces `None` without raising;
- [ ] missing or non-dict `plantDescription` produces nullable metadata fields;
- [ ] missing `displayName` follows the current fallback behavior;
- [ ] valid metadata that is present is preserved.

### Partial color structures

- [ ] missing color is safe;
- [ ] an empty color object is safe;
- [ ] omitted channels follow the current RGB helper contract;
- [ ] partial color data never interrupts the payload.

### Advice isolation

- [ ] plant entries do not inherit type-level `healthRecommendations`;
- [ ] a plant without advice keeps `advice` nullable.

Prefer extending existing plant shaping/advice tests and fixtures rather than creating broad new test infrastructure.

## Implementation brief

Suggested branch:

```text
test/partial-plant-payloads
```

Suggested PR title:

```text
test: cover partial plant payloads
```

Expected files:

```text
tests/test_sensor_plant_advice.py
tests/test_sensor.py
```

Use only the file or files required by the existing test layout.

## Constraints

- no production-code changes;
- no manifest or project-version changes;
- no README, FAQ, CHANGELOG, translations, services, migration, registry, diagnostics, or workflow changes;
- no raw live API fixture containing full recommendations, pictures, coordinates, or secrets;
- use the smallest synthetic payload reproducing the observed partial structures;
- keep code comments in English;
- preserve current behavior rather than introducing stricter parsing.

## Validation

```bash
python -m compileall -q sitecustomize.py custom_components tests
ruff check .
ruff format --check .
python -m pytest -q tests/test_sensor_plant_advice.py
python -m pytest -q tests/test_sensor.py
python -m pytest -q
```

CI must also pass hassfest, HACS validation, package checks, and CodeQL.

## PR evidence

The PR description must state:

- this is test-only;
- it is based on the completed 19-location live API audit;
- 17 of 19 locations contained missing plant `indexInfo` somewhere in the horizon;
- no runtime behavior, entity IDs, unique IDs, devices, subentries, forecast offsets, or migration logic changed;
- no new release candidate is required for this PR alone.

Use:

```text
Closes #212
```

## Release impact

Small test-only prerequisite for `3.0.0`. It does not require a new release candidate by itself.

### PR DESCRIPTION
**Pull request by eXPerience83**
## Summary

- add regression coverage for plant entries without `indexInfo`;
- cover missing plant metadata and partial color structures;
- preserve advice isolation between pollen types and plant entries.

## Evidence

The completed July 2026 live multi-region audit inspected 19 successful five-day responses and 95 location-days.

- 17 of 19 locations contained a plant without `indexInfo` somewhere in the horizon;
- 9 locations had optional plant structures that changed across forecast days;
- 0 invalid index values were observed;
- 0 API plant codes were missing from RC3 coordinator output;
- all 19 payloads were structurally compatible with RC3.

This PR locks the current tolerant behavior in without changing runtime code.

## Scope

Test-only change. No runtime, migration, registry, entity identity, subentry, forecast, service, translation, diagnostics, workflow, or release metadata changes.

## Validation

Local validation completed:

- `python -m compileall -q sitecustomize.py custom_components tests`
- `ruff check .`
- `ruff format --check .`
- `python -m pytest -q --noconftest tests/test_sensor_plant_advice.py`

The local container could not collect the Home Assistant-dependent full suite because Home Assistant and some test dependencies were unavailable. GitHub Actions completed successfully for:

- Lint;
- full tests;
- Package Test;
- hassfest;
- HACS validation.

Closes #212

## Summary by CodeRabbit

* **Tests**
  * Added regression coverage for handling incomplete plant data without disrupting coordinator output.
  * Verified safe defaults for missing index information and optional metadata.
  * Added coverage for missing, empty, partial, and zero-valued color data.

### CODE REVIEW

File: tests/test_sensor_plant_advice.py [hunk 1]
```diff
@@ -213,3 +213,232 @@ def test_plant_sensor_does_not_inherit_type_health_recommendations(
...
+    assert hazel["category"] is None
+    assert hazel["description"] is None
+    assert hazel["color_hex"] is None
+    assert hazel["color_rgb"] is None
+
+    assert "plants_oak" in data
+    assert data["plants_oak"]["value"] == 3
+    assert "type_tree" in data
+    assert data["type_tree"]["value"] == 2
+
```
[the remark this answers is not in the document]
Comment (eXPerience83) [reply tests/test_sensor_plant_advice.py:287]: Addressed in 8be7e0b. The partial-plant test now provides type-level health recommendations, verifies that they remain on the pollen type, and confirms that the plant without index data keeps `advice` as `None`.
